examples/barotropic_tide.jl

```diff
@@ -0,0 +1,285 @@
+                                  x = (-Lx/2, Lx/2),
+                                  z = (-H, 0),
+                                  halo = (4, 4),
+                                  topology = (Periodic, Flat, Bounded))
+
+# Now we can create the non-trivial bathymetry as an immersed boundary. We use `GridFittedBottom` that
+# gets as input either *(i)* a two-dimensional function whose arguments are the grid's native horizontal
+# coordinates and it returns the ``z`` of the bottom, or *(ii)* a two-dimensional array with the values
+# of ``z`` at the bottom at the cell centers.
+#
+# In this example we'd like to have a small Gaussian hill at the center of the domain.
+
+# ```math
+# h(x) = -H + h_0 \exp(-x^2 / 2σ^2)
+# ```
+
+h₀ = 50 # m
+width = 5kilometers
+hill(x) = h₀ * exp(-x^2 / 2width^2)
+bottom(x, y) = - H + hill(x)
+
+grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBottom(bottom))
+
+# The hill is small; here's how it looks (note that we don't plot all the way to the ocean surface).
+
+xC = xnodes(grid, Center())
+bottom_boundary = grid.immersed_boundary.bottom_height[1:Nx, 1]
+
+using CairoMakie
+
+fig = Figure(resolution = (700, 200))
+ax = Axis(fig[1, 1],
+          xlabel="x [km]",
+          ylabel="z [m]",
+          limits=((-Lx/2e3, Lx/2e3), (-H, -4H/5)))
+
+lines!(ax, xC/1e3, bottom_boundary)
+
+fig
+
+# Now we want to add a barotropic tide forcing. For example, to add the ``M_2`` tidal forcing
+# we need to add forcing in the ``u``-momentum equation of the form:
+# ```math
+# \partial_t u = \dotsb + F_0 \sin(\omega_2 t)
+# ```
+# where ``\omega_2 = 2π / T_2``, with ``T_2 = 12.421 \,\mathrm{hours}`` the period of the ``M_2`` tide.
+
+# The excursion parameter is a nondimensional number that expresses the ratio of the flow movement
+# due to the tide compared to the size of the width of the hill.
+#
+# ```math
+# \epsilon = \frac{U_{\mathrm{tidal}} / \omega_2}{\sigma}
+# ```
+# 
+# We prescribe the excursion parameter which, in turn, implies a tidal velocity ``U_{\mathrm{tidal}}``
+# which then allows us to determing the tidal forcing amplitude ``F_0``. For the last step, we
+# use Fourier decomposition on the inviscid, linearized momentum equations to determine the
+# flow response for a given tidal forcing. Doing so we get that for the sinusoidal forcing above,
+# the tidal velocity and tidal forcing amplitudes are related via:
+#
+# ```math
+# U_{\mathrm{tidal}} = \frac{\omega_2}{\omega_2^2 - f^2}F_0
+# ```
+#
+# Now we have the way to find the value of the tidal forcing amplitude that would correspond to a
+# given excursion parameter. The Coriolis frequency is needed, so we start by constructing a Coriolis on an ``f``-plane at the
+# mid-latitudes.
+
+coriolis = FPlane(latitude = -45)
+
+# Now we have everything we require to construct the tidal forcing given a value of the
+# excursion parameter.
+#
+T₂ = 12.421hours
+const ω₂ = 2π / T₂ # radians/sec
+
+ε = 0.25 # the excursion parameter
```

Comment:
```suggestion
ε = 0.25 # excursion parameter
```

we dont usually use varepsilon...

Reply:
I can use `\epsilon` :)
(it's just that ε is the default if I use the greek input from my keyboard...)

Reply:
Probably makes sense to be consistent, but if we want to change to varepsilon everywhere we should.

Reply:
\epsilon is great ;)